### PR TITLE
bolt: add tx_init_rbf and tx_ack_rbf codecs

### PR DESCRIPTION
**Pull request by devvaansh**
Add encoding/decoding support for two interactive transaction RBF messages from BOLT 2:

- **`tx_init_rbf`** (type 72): `channel_id` (32) + `locktime` (u32, 4) + `feerate` (u32, 4) — 40 bytes total
- **`tx_ack_rbf`** (type 73): `channel_id` (32) — 32 bytes total

Both messages are part of the interactive transaction protocol's RBF (Replace-By-Fee) flow, where `tx_init_rbf` initiates an RBF attempt and `tx_ack_rbf` acknowledges it.

## Changes

- **`smite/src/bolt/tx_init_rbf.rs`** — New codec: `TxInitRbf` struct with `encode()`/`decode()` and tests (roundtrip, field sizes, truncation for each field, empty payload)
- **`smite/src/bolt/tx_ack_rbf.rs`** — New codec: `TxAckRbf` struct with `encode()`/`decode()` and tests (roundtrip, field sizes, truncation, empty payload, trailing bytes)
- **`smite/src/bolt.rs`** — Wire both codecs into the `Message` enum with msg_type constants (`TX_INIT_RBF = 72`, `TX_ACK_RBF = 73`), `msg_type()`/`encode()`/`decode()` match arms, and Message-level roundtrip + type_values tests

## Testing

- All 261 bolt tests pass
- `cargo clippy` clean (0 warnings)
- `cargo fmt --all --check` clean

## Notes

TLV streams (`funding_output_contribution`, `require_confirmed_inputs`) are omitted for now, consistent with how existing codecs handle optional TLVs.

### CODE REVIEW

File: smite/src/bolt/tx_init_rbf.rs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+/// locktime and feerate that will apply to the replacement transaction.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TxInitRbf {
+    /// The channel ID.
+    pub channel_id: ChannelId,
+    /// The locktime for the replacement transaction.
+    pub locktime: u32,
+    /// The feerate for the replacement transaction (satoshis per kilo-weight).
+    pub feerate: u32,
+}
```
Comment (morehouse) [inline smite/src/bolt/tx_init_rbf.rs:43]: `tlvs` is missing.

File: smite/src/bolt/tx_ack_rbf.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+/// BOLT 2 `tx_ack_rbf` message (type 73).
+///
+/// Sent by the non-initiator to acknowledge an RBF attempt initiated by
+/// the peer via `tx_init_rbf`.  After this message, both sides begin a
+/// new interactive transaction construction for the replacement transaction.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TxAckRbf {
+    /// The channel ID.
+    pub channel_id: ChannelId,
+}
```
Comment (morehouse) [inline smite/src/bolt/tx_ack_rbf.rs:39]: `tlvs` is missing.

File: smite/src/bolt/tx_ack_rbf.rs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            TxAckRbf::decode(&[]),
+            Err(BoltError::Truncated {
+                expected: CHANNEL_ID_SIZE,
+                actual: 0
+            })
+        );
+    }
+
+    #[test]
+    fn decode_ignores_trailing_bytes() {
```
Comment (morehouse) [inline smite/src/bolt/tx_ack_rbf.rs]: Trailing bytes should be decoded as TLVs, not ignored.

**Review by morehouse — CHANGES_REQUESTED** [review]

File: smite/src/bolt/tx_ack_rbf.rs (hunk 3 above)
Comment (devvaansh) [reply smite/src/bolt/tx_ack_rbf.rs]: trailing bytes are now decoded as a TLV stream instead of being ignored.

File: smite/src/bolt/tx_ack_rbf.rs (hunk 2 above)
Comment (devvaansh) [reply smite/src/bolt/tx_ack_rbf.rs:39]: Added with tests.

File: smite/src/bolt/tx_init_rbf.rs (hunk 1 above)
Comment (devvaansh) [reply smite/src/bolt/tx_init_rbf.rs:43]: Added with tests.

File: smite/src/bolt/tx_ack_rbf.rs [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+const TLV_FUNDING_OUTPUT_CONTRIBUTION: u64 = 0;
+
+/// TLV type for require confirmed inputs.
+const TLV_REQUIRE_CONFIRMED_INPUTS: u64 = 2;
+
+/// Known even TLV types for `tx_ack_rbf`.
+const KNOWN_EVEN: &[u64] = &[
+    TLV_FUNDING_OUTPUT_CONTRIBUTION,
+    TLV_REQUIRE_CONFIRMED_INPUTS,
+];
```
Comment (morehouse) [inline smite/src/bolt/tx_ack_rbf.rs]: Nit: since there's only two even TLVs, perhaps we should just inline this list where it's used.

(also applies to the other codec)

File: smite/src/bolt/tx_ack_rbf.rs [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+        let funding_output_contribution =
+            if let Some(data) = stream.get(TLV_FUNDING_OUTPUT_CONTRIBUTION) {
+                if data.len() != 8 {
+                    return Err(BoltError::Truncated {
+                        expected: 8,
+                        actual: data.len(),
+                    });
+                }
+                let bytes: [u8; 8] = data.try_into().expect("length checked");
+                Some(i64::from_be_bytes(bytes))
```
Comment (morehouse) [inline smite/src/bolt/tx_ack_rbf.rs]: It may be worth adding a `WireFormat` impl for `i64` in this PR or as a follow-up.

File: smite/src/bolt/tx_ack_rbf.rs [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+    #[test]
+    fn decode_truncated_channel_id() {
+        assert_eq!(
+            TxAckRbf::decode(&[0x00; 20]),
+            Err(BoltError::Truncated {
+                expected: CHANNEL_ID_SIZE,
+                actual: 20
+            })
+        );
+    }
```
Comment (morehouse) [inline smite/src/bolt/tx_ack_rbf.rs:208]: We should also test a truncated `funding_output_contribution` for both codecs.

File: smite/src/bolt/tx_ack_rbf.rs (hunk 4 above)
Comment (devvaansh) [reply smite/src/bolt/tx_ack_rbf.rs]: Done, inlined the list directly at the [decode_with_known] call site in both codecs.

File: smite/src/bolt/tx_ack_rbf.rs (hunk 5 above)
Comment (devvaansh) [reply smite/src/bolt/tx_ack_rbf.rs]: Added impl_wire_format_int!(i64) in wire.rs and updated both codecs to use i64::read() in from_stream, replacing the manual from_be_bytes logic.

File: smite/src/bolt/tx_ack_rbf.rs (hunk 6 above)
Comment (devvaansh) [reply smite/src/bolt/tx_ack_rbf.rs:208]: Added decode_truncated_funding_output_contribution tests for both codecs.

File: smite/src/bolt/wire.rs [hunk 7]
```diff
@@ -60,6 +60,7 @@ impl_wire_format_int!(u8);
 impl_wire_format_int!(u16);
 impl_wire_format_int!(u32);
 impl_wire_format_int!(u64);
+impl_wire_format_int!(i64);
```
Comment (morehouse) [inline smite/src/bolt/wire.rs:63]: We also need to add tests for the new type, especially for negative values since that's new.

Also please make this change in its own commit that comes before the new codecs.

**Review by morehouse — COMMENTED** [review]
Looks good, we just need tests for the new `WireFormat` impl.

File: smite/src/bolt/wire.rs (hunk 7 above)
Comment (devvaansh) [reply smite/src/bolt/wire.rs:63]: i have  moved the i64 impl and tests into a separate first commit and added tests for reading, writing, truncation, and roundtrips, including negative values like -1, -100000, and i64::MIN, plus a test verifying the wire encoding of negatives.

File: smite/src/bolt/wire.rs [hunk 8]
```diff
@@ -357,6 +358,70 @@ mod tests {
         }
     }
 
+    #[test]
+    fn i64_read_valid() {
```
Comment (morehouse) [inline smite/src/bolt/wire.rs:362]: Let's also test that the cursor advances, for consistency with the other `read_valid` tests.

Also would be good to test `i64` max and min values.

**Review by morehouse — APPROVED** [review]

File: smite/src/bolt/wire.rs (hunk 8 above)
Comment (devvaansh) [reply smite/src/bolt/wire.rs:362]: yupp!! i have Added cursor-advancement check (two back-to-back reads) and i64 max and min cases.

**Review by morehouse — APPROVED** [review]
LGTM
